{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "b8399058",
   "metadata": {},
   "source": [
    "# Figures and analyses for propgage paper\n",
    "\n",
    "Figures and statistical tests appear in the order that they do in the manuscript. Supplementary figures are at the end.\n",
    "\n",
    "Note that we import a package from R for Hartigan's dip test, i.e. you will need a working installation of R configured to work with python to run that part of the notebook."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "007e01b7",
   "metadata": {},
   "source": [
    "### Import packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "3ec29cd7",
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "import math\n",
    "import re\n",
    "import string\n",
    "\n",
    "from PhiSpyAnalysis import theils_u, DateConverter, printmd\n",
    "from PhiSpyAnalysis import read_phages, read_gtdb, read_checkv, read_base_pp, read_categories, read_metadata, read_transposons\n",
    "from Bio import Phylo\n",
    "\n",
    "\n",
    "from scipy.stats import pearsonr, f_oneway, mannwhitneyu\n",
    "import scipy as scp\n",
    "from sklearn.linear_model import LinearRegression\n",
    "from sklearn.model_selection import train_test_split, LeaveOneGroupOut\n",
    "from sklearn import metrics\n",
    "import statsmodels.api as sm\n",
    "from statsmodels.formula.api import ols\n",
    "from statsmodels.stats.multicomp import pairwise_tukeyhsd, tukeyhsd, MultiComparison\n",
    "from statsmodels.multivariate.manova import MANOVA\n",
    "from sklearn import decomposition\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "# import rpy2.robjects as robjects\n",
    "# from rpy2.robjects.packages import importr\n",
    "# from rpy2.robjects import pandas2ri\n",
    "# from rpy2.robjects import Formula\n",
    "\n",
    "# for parsing collection dates\n",
    "from dateutil.parser import parse, ParserError\n",
    "import pytz\n",
    "\n",
    "import subprocess\n",
    "import gzip\n",
    "import pickle as pkl\n",
    "# import zstd\n",
    "\n",
    "# import function dc which converts a date to a year and decimal (e.g. 1/1/1900 is 1900.00)\n",
    "dc = DateConverter()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "86bb412a",
   "metadata": {},
   "source": [
    "### Import plot tools and figure settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c84cd303",
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/1l/n1k5w73d0gn1r0006j7fvt340000gn/T/ipykernel_31028/2838924867.py:32: MatplotlibDeprecationWarning: Support for setting an rcParam that expects a str value to a non-str value is deprecated since 3.5 and support will be removed two minor releases later.\n",
      "  mpl.rcParams['text.latex.preamble'] = [r'\\usepackage[cm]{sfmath}',r'\\usepackage{underscore}']\n"
     ]
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import matplotlib as mpl\n",
    "import matplotlib.gridspec as gridspec\n",
    "import matplotlib.ticker as mtick\n",
    "# from matplotlib.ticker import FuncFormatter\n",
    "# from matplotlib.ticker import ScalarFormatter\n",
    "formatter = mtick.ScalarFormatter(useMathText=True)\n",
    "formatter.set_scientific(True)\n",
    "from matplotlib.lines import Line2D\n",
    "from matplotlib.colors import LogNorm, Normalize\n",
    "import matplotlib.cm as cm\n",
    "from matplotlib.colors import Normalize\n",
    "from matplotlib import colors\n",
    "\n",
    "\n",
    "colourDict = dict(zip(['rose', 'indigo', 'sand', 'green', 'cyan', 'wine', 'teal', 'olive', 'purple', 'pale_grey', 'black'],\n",
    "                 ['#CC6677', '#332288', '#DDCC77', '#117733', '#88CCEE',\n",
    "                    '#882255', '#44AA99', '#999933', '#AA4499', '#DDDDDD',\n",
    "                    '#000000']))\n",
    "\n",
    "\n",
    "fig_width_pt = 410.0  # Get this from LaTeX using \\showthe\\columnwidth\n",
    "inches_per_pt = 1.0/72.27               # Convert pt to inch\n",
    "golden_mean = (np.sqrt(5)-1.0)/2.0         # Aesthetic ratio\n",
    "fig_width = fig_width_pt*inches_per_pt  # width in inches\n",
    "fig_height = fig_width*golden_mean       # height in inches\n",
    "fig_size =  [fig_width,fig_height]\n",
    "params = {'backend': 'ps','axes.labelsize': 8,'font.size': 8,\n",
    "          'legend.fontsize': 8,'xtick.labelsize': 8,'ytick.labelsize': 8,\n",
    "          'text.usetex': True,'figure.figsize': fig_size, 'axes.linewidth':0.3}\n",
    "mpl.rcParams['text.latex.preamble'] = [r'\\usepackage[cm]{sfmath}',r'\\usepackage{underscore}']\n",
    "mpl.rcParams['font.family'] = 'sans-serif'\n",
    "mpl.rcParams['font.sans-serif'] = 'cm'\n",
    "mpl.rcParams['text.usetex']=False\n",
    "mpl.rcParams['figure.dpi'] = 300\n",
    "\n",
    "mpl.rcParams.update(params)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9fed8488",
   "metadata": {},
   "source": [
    "### Some useful functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d9142bc6",
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "class SeabornFig2Grid():\n",
    "\n",
    "    def __init__(self, seaborngrid, fig,  subplot_spec):\n",
    "        self.fig = fig\n",
    "        self.sg = seaborngrid\n",
    "        self.subplot = subplot_spec\n",
    "        if isinstance(self.sg, sns.axisgrid.FacetGrid) or \\\n",
    "            isinstance(self.sg, sns.axisgrid.PairGrid):\n",
    "            self._movegrid()\n",
    "        elif isinstance(self.sg, sns.axisgrid.JointGrid):\n",
    "            self._movejointgrid()\n",
    "        self._finalize()\n",
    "\n",
    "    def _movegrid(self):\n",
    "        \"\"\" Move PairGrid or Facetgrid \"\"\"\n",
    "        self._resize()\n",
    "        n = self.sg.axes.shape[0]\n",
    "        m = self.sg.axes.shape[1]\n",
    "        self.subgrid = gridspec.GridSpecFromSubplotSpec(n,m, subplot_spec=self.subplot)\n",
    "        for i in range(n):\n",
    "            for j in range(m):\n",
    "                self._moveaxes(self.sg.axes[i,j], self.subgrid[i,j])\n",
    "\n",
    "    def _movejointgrid(self):\n",
    "        \"\"\" Move Jointgrid \"\"\"\n",
    "        h= self.sg.ax_joint.get_position().height\n",
    "        h2= self.sg.ax_marg_x.get_position().height\n",
    "        r = int(np.round(h/h2))\n",
    "        self._resize()\n",
    "        self.subgrid = gridspec.GridSpecFromSubplotSpec(r+1,r+1, subplot_spec=self.subplot)\n",
    "\n",
    "        self._moveaxes(self.sg.ax_joint, self.subgrid[1:, :-1])\n",
    "        self._moveaxes(self.sg.ax_marg_x, self.subgrid[0, :-1])\n",
    "        self._moveaxes(self.sg.ax_marg_y, self.subgrid[1:, -1])\n",
    "\n",
    "    def _moveaxes(self, ax, gs):\n",
    "        #https://stackoverflow.com/a/46906599/4124317\n",
    "        ax.remove()\n",
    "        ax.figure=self.fig\n",
    "        self.fig.axes.append(ax)\n",
    "        self.fig.add_axes(ax)\n",
    "        ax._subplotspec = gs\n",
    "        ax.set_position(gs.get_position(self.fig))\n",
    "        ax.set_subplotspec(gs)\n",
    "\n",
    "    def _finalize(self):\n",
    "        plt.close(self.sg.fig)\n",
    "        self.fig.canvas.mpl_connect(\"resize_event\", self._resize)\n",
    "        self.fig.canvas.draw()\n",
    "\n",
    "    def _resize(self, evt=None):\n",
    "        self.sg.fig.set_size_inches(self.fig.get_size_inches())\n",
    "        \n",
    "def saveZstd(object, filename):\n",
    "    bytesobject = pkl.dumps(object)\n",
    "    file = zstd.compress(bytesobject,20)\n",
    "    pkl.dump(file,open(filename,'wb'))\n",
    "    \n",
    "    \n",
    "def openZstd(filename):\n",
    "    fileObj = pkl.load(open(filename,'rb'))\n",
    "    decompObj = zstd.decompress(fileObj)\n",
    "    unpickleObj = pkl.loads(decompObj)\n",
    "    \n",
    "    return unpickleObj\n",
    "\n",
    "class MathTextSciFormatter(mtick.Formatter):\n",
    "    def __init__(self, fmt=\"%1.0e\"):\n",
    "        self.fmt = fmt\n",
    "    def __call__(self, x, pos=None):\n",
    "        s = self.fmt % x\n",
    "        decimal_point = '.'\n",
    "        positive_sign = '+'\n",
    "        tup = s.split('e')\n",
    "        significand = tup[0].rstrip(decimal_point)\n",
    "        sign = tup[1][0].replace(positive_sign, '')\n",
    "        exponent = tup[1][1:].lstrip('0')\n",
    "        if exponent:\n",
    "            exponent = '10^{%s%s}' % (sign, exponent)\n",
    "        if significand and exponent:\n",
    "            s =  r'%s{\\times}%s' % (significand, exponent)\n",
    "        else:\n",
    "            s =  r'%s%s' % (significand, exponent)\n",
    "        return \"${}$\".format(s)\n",
    "    \n",
    "def metagenomeStatus(x):\n",
    "    if 'metagenome' in x:\n",
    "        return \"Yes\" \n",
    "    else: \n",
    "        return \"No\"\n",
    "    \n",
    "def updateDate(x,magDateDict):\n",
    "#     c = .isnan()\n",
    "    if np.isnan(x['isolation_date']):\n",
    "        y = magDateDict.get(x['assembly_accession'])\n",
    "        return y\n",
    "    else:\n",
    "        y = x['isolation_date']\n",
    "        return y\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0ee6c6dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# from PhiSpyAnalysis import dtype_definitions \n",
    "# dtype_definitions.gtdb_dtypes()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92e6802b",
   "metadata": {},
   "source": [
    "# Import the datasets. \n",
    "\n",
    "We have two data sets: `small` is just 99 genomes and 1,561 phages and should run quickly for development. \n",
    "Max contig specifies the maximimum number of contigs. Set to 100 to ensure acceptable quality genomes.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8e5dedd1",
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "# use_small_data=False\n",
    "# maxContig = 100\n",
    "\n",
    "# phagesdf = read_phages(use_small_data=use_small_data, maxcontigs=maxContig)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2faefc18",
   "metadata": {},
   "source": [
    "### Read the metadata\n",
    "\n",
    "The metadata is from PATRIC. The phage data is from us.\n",
    "\n",
    "**NOTE:** Some of the PATRIC data refers to specific chromosomes/fragments in the GenBank file (e.g. plasmid, chromosome), but that may not equate to our predictions, because we have used the whole GenBank file. The PATRIC metadata is redundant for many fields, and so we just keep the first entry for each NCBI Assembly."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "fdb439b5",
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "# metadf = read_metadata(use_small_data=use_small_data)\n",
    "# metadf = metadf.drop_duplicates()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0395378",
   "metadata": {},
   "source": [
    "### Merge the dataframes\n",
    "\n",
    "First, select some columns we want to keep from PATRIC, and then merge the data frames."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "ad9400a2",
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "# acccol = 'assembly_accession'\n",
    "# interesting_cols = [acccol, 'isolation_site','taxon_id', 'isolation_country', 'latitude', 'longitude', 'altitude', 'depth',\n",
    "#                     'other_environmental', 'host_name', 'host_gender', 'host_age', 'host_health', \n",
    "#                     'body_sample_site', 'body_sample_subsite', 'other_clinical', 'gram_stain', 'cell_shape',\n",
    "#                     'motility', 'sporulation', 'temperature_range', 'optimal_temperature', 'salinity',\n",
    "#                     'oxygen_requirement', 'habitat', 'disease', 'isolation_date','refseq_accessions','pathovar']\n",
    "\n",
    "# few_interesting_cols = [acccol,  'isolation_country',  'gram_stain',  \n",
    "#                     'motility', 'sporulation', 'temperature_range', 'habitat', 'disease', 'isolation_date']\n",
    "\n",
    "# tempdf = metadf[interesting_cols]\n",
    "\n",
    "# phagemeta = pd.merge(tempdf, phagesdf, how='right', left_on=acccol, right_on=acccol)\n",
    "# phagemeta = phagemeta.dropna(subset=['Kept'])\n",
    "# phagemeta['normalise_by_prok_length'] = phagemeta['Genome length'].values - phagemeta['bp prophage'].values\n",
    "# phagemeta['Kept_normed'] = phagemeta['bp prophage'].values / phagemeta['normalise_by_prok_length'].values"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9f24a514",
   "metadata": {},
   "source": [
    "### Read the GTDB taxonomy data\n",
    "\n",
    "Genbank and PATRIC have NCBI taxonomy. Read in the GTDB data from the GTDB site.\n",
    "\n",
    "Note that the GTDB dump (read_gtdb()) does not contain all genomes from Genbank. We used GTDB-tk to get taxonomic assignments of all \"missing\" genomes in our dataset; these are imported in gtdbMissing and then merged."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b1ec5a69",
   "metadata": {},
   "outputs": [],
   "source": [
    "s = re.compile('^.__')\n",
    "# u = re.compile('_')\n",
    "def split_taxonomy(x):\n",
    "    p  = x.split(';')\n",
    "    if len(p) != 7:\n",
    "        sys.stderr.write(\"We have {len(p)} fields in {x}, but we expected 7 fields\")\n",
    "    # remove the prepended taxonomic rank name (e.g d__) and any remaining underscores before returning.\n",
    "    # return [u.sub(' ', l) for l in [s.sub('', m) for m in p] ]\n",
    "    return [s.sub('', m) for m in p]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32a0223f",
   "metadata": {},
   "source": [
    "##### NOTE!!!! Grabbing GTDB data\n",
    "For whatever reason there is a type error I can't resolve on the gtdb import function\n",
    "\n",
    "I am using the \"currentspeciesids\" for the rest of the notebook, which is from QC'd GTDB data\n",
    "\n",
    "If you want to use all of the genomes, there are about another 150k in the \"gtdbmissing\" list commented out below although some are duplicated with the new GTDB release, will clean up shortly\n",
    "\n",
    "however it appears many of these are of lower quality (maybe fragmented) so not good for finding 'nice' prophages which were dropped due to a lack of phage DNA\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4e1ef9b4",
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "tc = ['domain', 'phylum', 'class', 'order', 'family', 'genus', 'species']\n",
    "gtdb = pd.read_csv('../data/bac120_metadata_r207.tsv.gz', compression='gzip', sep='\\t',\n",
    "                  usecols=['ncbi_genbank_assembly_accession', 'gtdb_taxonomy'])\n",
    "\n",
    "tc = ['domain', 'phylum', 'class', 'order', 'family', 'genus', 'species']\n",
    "gtdb = pd.concat(\n",
    "    [gtdb, pd.DataFrame.from_records(gtdb['gtdb_taxonomy'].apply(split_taxonomy), columns=tc)],\n",
    "    axis=1)\n",
    "gtdb.rename(columns={'ncbi_genbank_assembly_accession':'assembly_accession'},inplace=True)\n",
    "taxameta = gtdb[['assembly_accession', 'domain', 'phylum', 'class', 'order', 'family', 'genus', 'species']]\n",
    "\n",
    "# gtdbMissing = pd.read_csv('../data/missing_taxa_april22.txt.gz',sep='\\t',compression='gzip',\n",
    "#                           names=['assembly_accession', 'domain', 'phylum', 'class', 'order', 'family', 'genus', 'species'],\n",
    "#                          index_col=None)\n",
    "\n",
    "# taxameta = taxameta.append(gtdbMissing)\n",
    "\n",
    "currentspeciesids = taxameta.copy()\n",
    "\n",
    "\n",
    "# taxameta = pd.merge(phagemeta,taxameta, on='assembly_accession', how='inner')\n",
    "\n",
    "# print(phagemeta.shape[0] - taxameta.shape[0],'genomes were in the PhiSpy dataset where we don\\'t have taxa information')\n",
    "# print('There are a total of', taxameta.shape[0], 'genomes in the study')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e64d885",
   "metadata": {},
   "source": [
    "## Create a balanced taxa sample from the ~45k dereplicated GTDB genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eef92fbc",
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "# # get the genomes on the 'leaves' of the GTDB taxa tree\n",
    "# tree = Phylo.read(\"../data/bac120_r202.tree\", \"newick\")\n",
    "# termList = []\n",
    "# for leaf in tree.get_terminals(): \n",
    "#     termList += [leaf.name]\n",
    "\n",
    "# # Get the species names for those genomes, create a new dataframe with accession and taxa name\n",
    "# balancedTax = pd.DataFrame()\n",
    "# balancedTax['assembly_accession'] = termList\n",
    "# balancedTax['assembly_accession'] = balancedTax['assembly_accession'].str.replace('RS_', '')\n",
    "# balancedTax['assembly_accession'] = balancedTax['assembly_accession'].str.replace('GB_', '')\n",
    "\n",
    "# gtdbDump = pd.read_csv('../data/bac120_taxonomy_r202.tsv', sep='\\t',header=None,\n",
    "#                          names=['assembly_accession','domain','phylum',\n",
    "#                                 'class','order','family','genus','species'])\n",
    "# gtdbDump['assembly_accession'] = gtdbDump['assembly_accession'].str.replace('RS_', '')\n",
    "# gtdbDump['assembly_accession'] = gtdbDump['assembly_accession'].str.replace('GB_', '')\n",
    "\n",
    "# speciesDict = dict(zip(gtdbDump['assembly_accession'],gtdbDump['species']))\n",
    "# balancedTax['species'] = balancedTax['assembly_accession'].map(speciesDict)\n",
    "\n",
    "# for col in balancedTax.columns:\n",
    "#     gtdbDump[col] = gtdbDump[col].str.replace('.__','',regex=True)\n",
    "# balancedTax = balancedTax[['assembly_accession']].merge(gtdbDump,on='assembly_accession',how='left')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "594d0dd7",
   "metadata": {},
   "source": [
    "There are ~45k unique species names in the GTDB tree. Let's check how many are in our <100 contig genbank data.\n",
    "\n",
    "Note that rerunning this cell will create a new random subset of the data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e501fa33",
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "# tmp = taxameta.sample(frac=1).reset_index(drop=True).drop_duplicates(subset='species',keep='first')\n",
    "# tmp = tmp.merge(balancedTax,on='species',how='right')\n",
    "\n",
    "# # Create a list of dereplicated genomes with the correct number of sampled genomes\n",
    "# newickList = list(tmp['assembly_accession_x'])\n",
    "# newickList = [item for item in newickList if not(pd.isnull(item)) == True]\n",
    "\n",
    "\n",
    "# print('We have',len(newickList),'species in our <100 contig genbank data')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1c7e3bb0",
   "metadata": {},
   "source": [
    "### Assembly summaries"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "504973b2",
   "metadata": {},
   "source": [
    "Use the genbank assembly summary to get additional metadata e.g. MAG status, as well as isolation date (proxied as sample date)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "91013e57",
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/1l/n1k5w73d0gn1r0006j7fvt340000gn/T/ipykernel_7623/2691450898.py:1: DtypeWarning: Columns (21) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  assembly_summary = pd.read_csv('../data/assembly_summary.txt.gz', compression='gzip',sep='\\t',skiprows=1)\n"
     ]
    }
   ],
   "source": [
    "assembly_summary = pd.read_csv('../data/assembly_summary.txt.gz', compression='gzip',sep='\\t',skiprows=1)\n",
    "assembly_summary.rename(columns={'# assembly_accession':'assembly_accession'},inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a926a0c",
   "metadata": {},
   "source": [
    "#### Update dates and MAG status in the Genbank summary file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "27d0a80d",
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/1l/n1k5w73d0gn1r0006j7fvt340000gn/T/ipykernel_7623/4175757948.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  assembly_summarySub.loc[:,'excluded_from_refseq'] = assembly_summarySub.loc[:,'excluded_from_refseq'].astype(str)\n",
      "/var/folders/1l/n1k5w73d0gn1r0006j7fvt340000gn/T/ipykernel_7623/4175757948.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  assembly_summarySub.loc[:,'MAG status'] = assembly_summarySub['excluded_from_refseq'].apply(lambda x: metagenomeStatus(x))\n",
      "/var/folders/1l/n1k5w73d0gn1r0006j7fvt340000gn/T/ipykernel_7623/4175757948.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  assembly_summarySub.loc[:,'isolation_date'] = assembly_summarySub.seq_rel_date.apply(dc.convert_date)\n"
     ]
    }
   ],
   "source": [
    "assembly_summarySub = assembly_summary[['assembly_accession','seq_rel_date','excluded_from_refseq']]\n",
    "assembly_summarySub.loc[:,'excluded_from_refseq'] = assembly_summarySub.loc[:,'excluded_from_refseq'].astype(str)\n",
    "assembly_summarySub.loc[:,'MAG status'] = assembly_summarySub['excluded_from_refseq'].apply(lambda x: metagenomeStatus(x))\n",
    "assembly_summarySub.loc[:,'isolation_date'] = assembly_summarySub.seq_rel_date.apply(dc.convert_date)\n",
    "assembly_summaryMag = assembly_summarySub[assembly_summarySub['MAG status'] == \"Yes\"]\n",
    "\n",
    "# # Create lists of genomes which are MAGs, and the dates they were \"isolated\" (sampled)\n",
    "magAccs = list(assembly_summaryMag['assembly_accession'])\n",
    "magDateDict = dict(zip(assembly_summarySub['assembly_accession'],assembly_summarySub['isolation_date']))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "733e7c6b",
   "metadata": {},
   "source": [
    "### Find good candidates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "281bae8d",
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>Contigs &gt; 5kb</th>\n",
       "      <th>Genome Contigs</th>\n",
       "      <th>Number of Coding Sequences</th>\n",
       "      <th>Too short</th>\n",
       "      <th>Not enough phage hits</th>\n",
       "      <th>Kept</th>\n",
       "      <th>Note</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_019958295.1</td>\n",
       "      <td>Buchnera aphidicola strain Tabriz.4 Contig1000...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1224</td>\n",
       "      <td>1075</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Too few ORFs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_900128725.1</td>\n",
       "      <td>Buchnera aphidicola strain BCifornacula genome...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>None of the features have protein sequences</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>Dictyoglomus thermophilum strain PYS_80_B</td>\n",
       "      <td>14.0</td>\n",
       "      <td>23</td>\n",
       "      <td>1876</td>\n",
       "      <td>8</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>Complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_008015785.1</td>\n",
       "      <td>Methylophilus methylotrophus isolate Bin_42_2 ...</td>\n",
       "      <td>94.0</td>\n",
       "      <td>94</td>\n",
       "      <td>1148</td>\n",
       "      <td>13</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>Complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_001735525.1</td>\n",
       "      <td>Shewanella colwelliana strain CSB03KR, whole g...</td>\n",
       "      <td>33.0</td>\n",
       "      <td>52</td>\n",
       "      <td>4040</td>\n",
       "      <td>23</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>Complete</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  assembly_accession                                        Genome Name  \\\n",
       "0    GCA_019958295.1  Buchnera aphidicola strain Tabriz.4 Contig1000...   \n",
       "1    GCA_900128725.1  Buchnera aphidicola strain BCifornacula genome...   \n",
       "2    GCA_008244535.1          Dictyoglomus thermophilum strain PYS_80_B   \n",
       "3    GCA_008015785.1  Methylophilus methylotrophus isolate Bin_42_2 ...   \n",
       "4    GCA_001735525.1  Shewanella colwelliana strain CSB03KR, whole g...   \n",
       "\n",
       "   Contigs > 5kb  Genome Contigs  Number of Coding Sequences  Too short  \\\n",
       "0            1.0            1224                        1075          0   \n",
       "1            2.0               3                           0          0   \n",
       "2           14.0              23                        1876          8   \n",
       "3           94.0              94                        1148         13   \n",
       "4           33.0              52                        4040         23   \n",
       "\n",
       "   Not enough phage hits  Kept                                         Note  \n",
       "0                      0     0                                 Too few ORFs  \n",
       "1                      0     0  None of the features have protein sequences  \n",
       "2                      5     4                                     Complete  \n",
       "3                     22     0                                     Complete  \n",
       "4                      4     3                                     Complete  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read the genome phage stats file\n",
    "phagestatsdf = pd.read_csv('../data/phage_stats.20220620.tsv', sep='\\t')\n",
    "phagestatsdf.rename(columns={'GENOMEID':'assembly_accession'},inplace=True)\n",
    "phagestatsdf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "280ced23",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>phispy contig</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>length</th>\n",
       "      <th>ngenes</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>VTFL01000006.1</td>\n",
       "      <td>337</td>\n",
       "      <td>34176</td>\n",
       "      <td>33840</td>\n",
       "      <td>27</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>VTFL01000004.1</td>\n",
       "      <td>75050</td>\n",
       "      <td>95353</td>\n",
       "      <td>20304</td>\n",
       "      <td>25</td>\n",
       "      <td>Kept</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>VTFL01000001.1</td>\n",
       "      <td>205725</td>\n",
       "      <td>224319</td>\n",
       "      <td>18595</td>\n",
       "      <td>23</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>VTFL01000001.1</td>\n",
       "      <td>71619</td>\n",
       "      <td>92803</td>\n",
       "      <td>21185</td>\n",
       "      <td>23</td>\n",
       "      <td>Kept</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>VTFL01000004.1</td>\n",
       "      <td>132141</td>\n",
       "      <td>156532</td>\n",
       "      <td>24392</td>\n",
       "      <td>22</td>\n",
       "      <td>Kept</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  assembly_accession   phispy contig   start    stop  length  ngenes  \\\n",
       "0    GCA_008244535.1  VTFL01000006.1     337   34176   33840      27   \n",
       "1    GCA_008244535.1  VTFL01000004.1   75050   95353   20304      25   \n",
       "2    GCA_008244535.1  VTFL01000001.1  205725  224319   18595      23   \n",
       "3    GCA_008244535.1  VTFL01000001.1   71619   92803   21185      23   \n",
       "4    GCA_008244535.1  VTFL01000004.1  132141  156532   24392      22   \n",
       "\n",
       "                                              notes  \n",
       "0  Dropped. No genes were identified as phage genes  \n",
       "1                                              Kept  \n",
       "2  Dropped. No genes were identified as phage genes  \n",
       "3                                              Kept  \n",
       "4                                              Kept  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read the phispy contig metadata file (stats per individual phage)\n",
    "allphagedf = pd.read_csv('../data/phage_locations_20220620.tsv', sep='\\t', header=None)\n",
    "allphagedf.rename(columns={0:'assembly_accession',1:'phispy contig',2:'start',3:'stop',4:'length',5:'ngenes',6:'notes'},inplace=True)\n",
    "allphagedf.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "44eac918",
   "metadata": {},
   "source": [
    "#### Get the total length of phage (\"Kept\") DNA in the genome and substitute zero back in if all of them are dropped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "88924c42",
   "metadata": {},
   "outputs": [],
   "source": [
    "# group the df to split the kept vs not phage dna\n",
    "groupeddf = allphagedf.copy() # duplicate\n",
    "groupeddf = allphagedf.groupby(['assembly_accession','notes']).sum()  # sum for total bp of phage DNA per category"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "60a41beb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/1l/n1k5w73d0gn1r0006j7fvt340000gn/T/ipykernel_7623/461416556.py:15: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  totalphagecontent = groupeddf.append(allrowdf.loc[missing_index, :]).reset_index()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000003135.1</td>\n",
       "      <td>48918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000003645.1</td>\n",
       "      <td>40298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000003925.1</td>\n",
       "      <td>268087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000003955.1</td>\n",
       "      <td>166292</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000005825.2</td>\n",
       "      <td>93419</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  assembly_accession  length\n",
       "0    GCA_000003135.1   48918\n",
       "1    GCA_000003645.1   40298\n",
       "2    GCA_000003925.1  268087\n",
       "3    GCA_000003955.1  166292\n",
       "4    GCA_000005825.2   93419"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# iterate over the genomes which have \"Kept\" phage \n",
    "allgenomes = list(allphagedf['assembly_accession'].unique())\n",
    "allnotes = ['Kept']\n",
    "iterables = [allgenomes, allnotes]\n",
    "\n",
    "# set up a df to append a length of zero whenever there are no \"kept\" phage\n",
    "allrowindex = pd.MultiIndex.from_product(iterables, names=[\"assembly_accession\", \"notes\"])\n",
    "allrowdf = pd.DataFrame(0, columns=['length'], index=allrowindex)\n",
    "\n",
    "# get indices where \"Kept\" is missing\n",
    "already_present_index = groupeddf.index.intersection(allrowdf.index)\n",
    "missing_index = allrowdf.index.difference(groupeddf.index)\n",
    "\n",
    "# Append total phage length for all genomes in dataset, with 0 whenever there are no kept phage\n",
    "totalphagecontent = groupeddf.append(allrowdf.loc[missing_index, :]).reset_index()\n",
    "totalphagecontent = totalphagecontent[totalphagecontent['notes'] == 'Kept'].reset_index(drop=True)\n",
    "totalphagecontent = totalphagecontent[['assembly_accession','length']]\n",
    "totalphagecontent.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "9c5ab75b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(945551, 10)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge the length info back to our summary metadata frame\n",
    "phagestatsdf = phagestatsdf.merge(totalphagecontent,on='assembly_accession', how='inner')\n",
    "phagestatsdf.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb5bd563",
   "metadata": {},
   "source": [
    "### Merge the new phage stats data with NCBI assembly dump"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "87bec95b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(949933, 25)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assembly_summary.columns\n",
    "assembly_summary['excluded_from_refseq'] = assembly_summary['excluded_from_refseq'].astype(str)\n",
    "assembly_summary.loc[:,'MAG status'] = assembly_summary['excluded_from_refseq'].apply(lambda x: metagenomeStatus(x))\n",
    "\n",
    "\n",
    "rankedlineagedf = pd.read_csv('../data/rankedlineage.dmp', sep='\\t', \n",
    "                              names=['taxid', 'strain','ncbispecies','genus','order',\n",
    "                                    'family','class','phylum','superphylum','kingdom'], low_memory=False)\n",
    "rankedlineagedf = rankedlineagedf[rankedlineagedf['kingdom'] == 'Bacteria']\n",
    "assembly_summary = assembly_summary.merge(rankedlineagedf[['taxid','ncbispecies','strain']], on='taxid',how='left')\n",
    "\n",
    "assembly_summary['ncbispecies'].fillna(assembly_summary['strain'], inplace=True)\n",
    "\n",
    "assembly_summary.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "41dee62f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>Contigs &gt; 5kb</th>\n",
       "      <th>Genome Contigs</th>\n",
       "      <th>Number of Coding Sequences</th>\n",
       "      <th>Too short</th>\n",
       "      <th>Not enough phage hits</th>\n",
       "      <th>Kept</th>\n",
       "      <th>Note</th>\n",
       "      <th>length</th>\n",
       "      <th>...</th>\n",
       "      <th>MAG status</th>\n",
       "      <th>ncbispecies</th>\n",
       "      <th>strain</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_008244535.1</td>\n",
       "      <td>Dictyoglomus thermophilum strain PYS_80_B</td>\n",
       "      <td>14.0</td>\n",
       "      <td>23</td>\n",
       "      <td>1876</td>\n",
       "      <td>8</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>Complete</td>\n",
       "      <td>79800</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Dictyoglomus thermophilum</td>\n",
       "      <td>Dictyoglomus thermophilum</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Dictyoglomota</td>\n",
       "      <td>Dictyoglomia</td>\n",
       "      <td>Dictyoglomales</td>\n",
       "      <td>Dictyoglomaceae</td>\n",
       "      <td>Dictyoglomus</td>\n",
       "      <td>Dictyoglomus thermophilum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_008015785.1</td>\n",
       "      <td>Methylophilus methylotrophus isolate Bin_42_2 ...</td>\n",
       "      <td>94.0</td>\n",
       "      <td>94</td>\n",
       "      <td>1148</td>\n",
       "      <td>13</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>Complete</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Methylophilus methylotrophus</td>\n",
       "      <td>Methylophilus methylotrophus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_001735525.1</td>\n",
       "      <td>Shewanella colwelliana strain CSB03KR, whole g...</td>\n",
       "      <td>33.0</td>\n",
       "      <td>52</td>\n",
       "      <td>4040</td>\n",
       "      <td>23</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>Complete</td>\n",
       "      <td>83562</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Shewanella colwelliana</td>\n",
       "      <td>Shewanella colwelliana</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Shewanellaceae</td>\n",
       "      <td>Shewanella</td>\n",
       "      <td>Shewanella colwelliana</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_003044255.1</td>\n",
       "      <td>Shewanella putrefaciens strain WS13 chromosome...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>4407</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>Complete</td>\n",
       "      <td>63450</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Shewanella putrefaciens</td>\n",
       "      <td>Shewanella putrefaciens</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Shewanellaceae</td>\n",
       "      <td>Shewanella</td>\n",
       "      <td>Shewanella putrefaciens</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_003315425.1</td>\n",
       "      <td>Shewanella putrefaciens strain 97 Ga0227277_10...</td>\n",
       "      <td>36.0</td>\n",
       "      <td>45</td>\n",
       "      <td>4323</td>\n",
       "      <td>22</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>Complete</td>\n",
       "      <td>76951</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Shewanella putrefaciens</td>\n",
       "      <td>Shewanella putrefaciens</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>Shewanellaceae</td>\n",
       "      <td>Shewanella</td>\n",
       "      <td>Shewanella sp002966515</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  assembly_accession                                        Genome Name  \\\n",
       "0    GCA_008244535.1          Dictyoglomus thermophilum strain PYS_80_B   \n",
       "1    GCA_008015785.1  Methylophilus methylotrophus isolate Bin_42_2 ...   \n",
       "2    GCA_001735525.1  Shewanella colwelliana strain CSB03KR, whole g...   \n",
       "3    GCA_003044255.1  Shewanella putrefaciens strain WS13 chromosome...   \n",
       "4    GCA_003315425.1  Shewanella putrefaciens strain 97 Ga0227277_10...   \n",
       "\n",
       "   Contigs > 5kb  Genome Contigs  Number of Coding Sequences  Too short  \\\n",
       "0           14.0              23                        1876          8   \n",
       "1           94.0              94                        1148         13   \n",
       "2           33.0              52                        4040         23   \n",
       "3            1.0               1                        4407         16   \n",
       "4           36.0              45                        4323         22   \n",
       "\n",
       "   Not enough phage hits  Kept      Note  length  ... MAG status  \\\n",
       "0                      5     4  Complete   79800  ...         No   \n",
       "1                     22     0  Complete       0  ...        Yes   \n",
       "2                      4     3  Complete   83562  ...         No   \n",
       "3                      8     1  Complete   63450  ...         No   \n",
       "4                      3     5  Complete   76951  ...         No   \n",
       "\n",
       "                    ncbispecies                        strain    domain  \\\n",
       "0     Dictyoglomus thermophilum     Dictyoglomus thermophilum  Bacteria   \n",
       "1  Methylophilus methylotrophus  Methylophilus methylotrophus       NaN   \n",
       "2        Shewanella colwelliana        Shewanella colwelliana  Bacteria   \n",
       "3       Shewanella putrefaciens       Shewanella putrefaciens  Bacteria   \n",
       "4       Shewanella putrefaciens       Shewanella putrefaciens  Bacteria   \n",
       "\n",
       "           phylum                class             order           family  \\\n",
       "0   Dictyoglomota         Dictyoglomia    Dictyoglomales  Dictyoglomaceae   \n",
       "1             NaN                  NaN               NaN              NaN   \n",
       "2  Proteobacteria  Gammaproteobacteria  Enterobacterales   Shewanellaceae   \n",
       "3  Proteobacteria  Gammaproteobacteria  Enterobacterales   Shewanellaceae   \n",
       "4  Proteobacteria  Gammaproteobacteria  Enterobacterales   Shewanellaceae   \n",
       "\n",
       "          genus                    species  \n",
       "0  Dictyoglomus  Dictyoglomus thermophilum  \n",
       "1           NaN                        NaN  \n",
       "2    Shewanella     Shewanella colwelliana  \n",
       "3    Shewanella    Shewanella putrefaciens  \n",
       "4    Shewanella     Shewanella sp002966515  \n",
       "\n",
       "[5 rows x 41 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### Merge on NCBI assembly summary\n",
    "mergedf = phagestatsdf.merge(assembly_summary, on='assembly_accession')\n",
    "mergedf = mergedf.merge(currentspeciesids,on='assembly_accession',how='left')\n",
    "mergedf.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d3f26170",
   "metadata": {},
   "source": [
    "## Find worthy candidates\n",
    "\n",
    "choose specs, defaults listed below\n",
    "\n",
    "I didn't filter on \"Complete\" as these filters made it redundant\n",
    "\n",
    "1.\t1 contig in the genome\n",
    "2.\tover 2000 coding genes in the genome\n",
    "3.\tCandidatus in the NCBI species name (did the job with minimal complexity, happy to try and be smarter)\n",
    "4.\tTake the rarest 25 phyla from the subsequent filtered list (the top few phyla have many hundreds of genomes)\n",
    "5.\tProphage needs to be > 40kb \n",
    "6.\tPotential phage “Dropped” due to no phage genes identified\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "451362e6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10148 candidatus genomes with 1 contigs in genome and 2000 cds\n"
     ]
    }
   ],
   "source": [
    "# top100specieslist = list(mergedf['organism_name'].value_counts().head(200).index)\n",
    "contigsInGenome = 1\n",
    "cdsInGenome = 2000\n",
    "stringToFilter = 'Candidatus' \n",
    "rarestPhyla = 25\n",
    "minProphageLength = 40000\n",
    "\n",
    "\n",
    "# could use a phylum list filter (phylum is in ... instead here of candidatus in NCBI species name)\n",
    "candidatuslist = list(mergedf[mergedf['ncbispecies'].str.contains(stringToFilter, na=False)]['assembly_accession'])\n",
    "\n",
    "# start filtering the dataset down\n",
    "subdf = mergedf[~mergedf['organism_name'].isin(candidatuslist)]\n",
    "subdf = subdf[(subdf['Genome Contigs'] <= contigsInGenome) & (subdf['Number of Coding Sequences'] > cdsInGenome)]\n",
    "print(subdf.shape[0], 'candidatus genomes with', contigsInGenome, 'contigs in genome and',cdsInGenome,'cds')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "b61c31af",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "121 genomes from the rarest phyla\n",
      "46 genomes with \"dropped\" phages over 40000 bp\n"
     ]
    }
   ],
   "source": [
    "# grab rare phyla\n",
    "rarephyla = subdf['phylum'].value_counts()[-rarestPhyla:].index\n",
    "subdf = subdf[subdf['phylum'].isin(rarephyla)]\n",
    "print(subdf.shape[0], 'genomes from the rarest phyla')\n",
    "\n",
    "# filter based on length\n",
    "genomesofinterestList = list(subdf['assembly_accession'])\n",
    "interrogateSubsetDf = allphagedf[(allphagedf['assembly_accession'].isin(genomesofinterestList)) & \n",
    "                                 (allphagedf['length'] > minProphageLength) & \n",
    "                                 (allphagedf['notes'] == 'Dropped. No genes were identified as phage genes')].reset_index(drop=True)\n",
    "\n",
    "print(interrogateSubsetDf.shape[0], 'genomes with \"dropped\" phages over', minProphageLength, 'bp')\n",
    "\n",
    "summaryFilteredGenomeTable = mergedf[mergedf['assembly_accession'].isin(genomesofinterestList)]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a99666d1",
   "metadata": {},
   "source": [
    "#### Save to file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "66c2bb43",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>Contigs &gt; 5kb</th>\n",
       "      <th>Genome Contigs</th>\n",
       "      <th>Number of Coding Sequences</th>\n",
       "      <th>Too short</th>\n",
       "      <th>Not enough phage hits</th>\n",
       "      <th>Kept</th>\n",
       "      <th>Note</th>\n",
       "      <th>length</th>\n",
       "      <th>...</th>\n",
       "      <th>MAG status</th>\n",
       "      <th>ncbispecies</th>\n",
       "      <th>strain</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>72261</th>\n",
       "      <td>GCA_004801255.1</td>\n",
       "      <td>Desulfovibrio desulfuricans strain IC1 chromos...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2680</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>Complete</td>\n",
       "      <td>67806</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Desulfovibrio desulfuricans</td>\n",
       "      <td>Desulfovibrio desulfuricans</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Desulfobacterota_I</td>\n",
       "      <td>Desulfovibrionia</td>\n",
       "      <td>Desulfovibrionales</td>\n",
       "      <td>Desulfovibrionaceae</td>\n",
       "      <td>Desulfovibrio</td>\n",
       "      <td>Desulfovibrio desulfuricans_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72354</th>\n",
       "      <td>GCA_900116045.1</td>\n",
       "      <td>Desulfovibrio piger isolate FI11049 genome ass...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2535</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>Complete</td>\n",
       "      <td>176572</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Desulfovibrio piger</td>\n",
       "      <td>Desulfovibrio piger</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Desulfobacterota_I</td>\n",
       "      <td>Desulfovibrionia</td>\n",
       "      <td>Desulfovibrionales</td>\n",
       "      <td>Desulfovibrionaceae</td>\n",
       "      <td>Desulfovibrio</td>\n",
       "      <td>Desulfovibrio piger_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72454</th>\n",
       "      <td>GCA_000691605.1</td>\n",
       "      <td>Bdellovibrio bacteriovorus strain 109J, comple...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>3589</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Complete</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Bdellovibrio bacteriovorus</td>\n",
       "      <td>Bdellovibrio bacteriovorus</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bdellovibrionota</td>\n",
       "      <td>Bdellovibrionia</td>\n",
       "      <td>Bdellovibrionales</td>\n",
       "      <td>Bdellovibrionaceae</td>\n",
       "      <td>Bdellovibrio</td>\n",
       "      <td>Bdellovibrio bacteriovorus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72458</th>\n",
       "      <td>GCA_002208115.1</td>\n",
       "      <td>Bdellovibrio bacteriovorus strain SSB218315 ch...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>3579</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Complete</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Bdellovibrio bacteriovorus</td>\n",
       "      <td>Bdellovibrio bacteriovorus</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bdellovibrionota</td>\n",
       "      <td>Bdellovibrionia</td>\n",
       "      <td>Bdellovibrionales</td>\n",
       "      <td>Bdellovibrionaceae</td>\n",
       "      <td>Bdellovibrio</td>\n",
       "      <td>Bdellovibrio bacteriovorus_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72459</th>\n",
       "      <td>GCA_002872415.1</td>\n",
       "      <td>Bacteriovorax stolpii strain DSM 12778 chromos...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>3742</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>Complete</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Bacteriovorax stolpii</td>\n",
       "      <td>Bacteriovorax stolpii</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bdellovibrionota</td>\n",
       "      <td>Bacteriovoracia</td>\n",
       "      <td>Bacteriovoracales</td>\n",
       "      <td>Bacteriovoracaceae</td>\n",
       "      <td>Bacteriovorax</td>\n",
       "      <td>Bacteriovorax stolpii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647687</th>\n",
       "      <td>GCA_013004105.1</td>\n",
       "      <td>Gemmatimonas sp. TET16 chromosome, complete ge...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>4390</td>\n",
       "      <td>29</td>\n",
       "      <td>8</td>\n",
       "      <td>5</td>\n",
       "      <td>Complete</td>\n",
       "      <td>131511</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Gemmatimonas groenlandica</td>\n",
       "      <td>Gemmatimonas groenlandica</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Gemmatimonadota</td>\n",
       "      <td>Gemmatimonadetes</td>\n",
       "      <td>Gemmatimonadales</td>\n",
       "      <td>Gemmatimonadaceae</td>\n",
       "      <td>Gemmatimonas</td>\n",
       "      <td>Gemmatimonas sp013004105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647896</th>\n",
       "      <td>GCA_015709495.1</td>\n",
       "      <td>Chthonomonadaceae bacterium isolate H1_ARM1 ch...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2631</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Complete</td>\n",
       "      <td>24691</td>\n",
       "      <td>...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Chthonomonadaceae bacterium</td>\n",
       "      <td>Chthonomonadaceae bacterium</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Armatimonadota</td>\n",
       "      <td>Fimbriimonadia</td>\n",
       "      <td>Fimbriimonadales</td>\n",
       "      <td>Fimbriimonadaceae</td>\n",
       "      <td>H1-ARM1</td>\n",
       "      <td>H1-ARM1 sp001567425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>648315</th>\n",
       "      <td>GCA_013752535.1</td>\n",
       "      <td>Bdellovibrio sp. KM01 chromosome, complete gen...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>3769</td>\n",
       "      <td>32</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>Complete</td>\n",
       "      <td>46433</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Bdellovibrio sp. KM01</td>\n",
       "      <td>Bdellovibrio sp. KM01</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bdellovibrionota</td>\n",
       "      <td>Bdellovibrionia</td>\n",
       "      <td>Bdellovibrionales</td>\n",
       "      <td>Bdellovibrionaceae</td>\n",
       "      <td>Bdellovibrio</td>\n",
       "      <td>Bdellovibrio sp013752535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>648637</th>\n",
       "      <td>GCA_002563855.1</td>\n",
       "      <td>Thermoflexus hugenholtzii strain G233 Ga007014...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2656</td>\n",
       "      <td>13</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>Complete</td>\n",
       "      <td>71117</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Chloroflexi bacterium G233</td>\n",
       "      <td>Chloroflexi bacterium G233</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Chloroflexota</td>\n",
       "      <td>Dehalococcoidia</td>\n",
       "      <td>Tepidiformales</td>\n",
       "      <td>Tepidiformaceae</td>\n",
       "      <td>Tepidiforma</td>\n",
       "      <td>Tepidiforma sp002563855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650081</th>\n",
       "      <td>GCA_015775515.1</td>\n",
       "      <td>Candidatus Atribacteria bacterium strain RT761...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2759</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>Complete</td>\n",
       "      <td>115254</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Candidatus Atribacteria bacterium</td>\n",
       "      <td>Candidatus Atribacteria bacterium</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Atribacterota</td>\n",
       "      <td>Atribacteria</td>\n",
       "      <td>Atribacterales</td>\n",
       "      <td>Atribacteraceae</td>\n",
       "      <td>Atribacter</td>\n",
       "      <td>Atribacter laminatus</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       assembly_accession                                        Genome Name  \\\n",
       "72261     GCA_004801255.1  Desulfovibrio desulfuricans strain IC1 chromos...   \n",
       "72354     GCA_900116045.1  Desulfovibrio piger isolate FI11049 genome ass...   \n",
       "72454     GCA_000691605.1  Bdellovibrio bacteriovorus strain 109J, comple...   \n",
       "72458     GCA_002208115.1  Bdellovibrio bacteriovorus strain SSB218315 ch...   \n",
       "72459     GCA_002872415.1  Bacteriovorax stolpii strain DSM 12778 chromos...   \n",
       "...                   ...                                                ...   \n",
       "647687    GCA_013004105.1  Gemmatimonas sp. TET16 chromosome, complete ge...   \n",
       "647896    GCA_015709495.1  Chthonomonadaceae bacterium isolate H1_ARM1 ch...   \n",
       "648315    GCA_013752535.1  Bdellovibrio sp. KM01 chromosome, complete gen...   \n",
       "648637    GCA_002563855.1  Thermoflexus hugenholtzii strain G233 Ga007014...   \n",
       "650081    GCA_015775515.1  Candidatus Atribacteria bacterium strain RT761...   \n",
       "\n",
       "        Contigs > 5kb  Genome Contigs  Number of Coding Sequences  Too short  \\\n",
       "72261             1.0               1                        2680         10   \n",
       "72354             1.0               1                        2535         10   \n",
       "72454             1.0               1                        3589          7   \n",
       "72458             1.0               1                        3579         11   \n",
       "72459             1.0               1                        3742          5   \n",
       "...               ...             ...                         ...        ...   \n",
       "647687            1.0               1                        4390         29   \n",
       "647896            1.0               1                        2631         23   \n",
       "648315            1.0               1                        3769         32   \n",
       "648637            1.0               1                        2656         13   \n",
       "650081            1.0               1                        2759          4   \n",
       "\n",
       "        Not enough phage hits  Kept      Note  length  ... MAG status  \\\n",
       "72261                       6     3  Complete   67806  ...         No   \n",
       "72354                       2     4  Complete  176572  ...         No   \n",
       "72454                       1     0  Complete       0  ...         No   \n",
       "72458                       0     0  Complete       0  ...         No   \n",
       "72459                       6     0  Complete       0  ...         No   \n",
       "...                       ...   ...       ...     ...  ...        ...   \n",
       "647687                      8     5  Complete  131511  ...         No   \n",
       "647896                      1     1  Complete   24691  ...        Yes   \n",
       "648315                      4     2  Complete   46433  ...         No   \n",
       "648637                      4     2  Complete   71117  ...         No   \n",
       "650081                     10     4  Complete  115254  ...         No   \n",
       "\n",
       "                              ncbispecies                             strain  \\\n",
       "72261         Desulfovibrio desulfuricans        Desulfovibrio desulfuricans   \n",
       "72354                 Desulfovibrio piger                Desulfovibrio piger   \n",
       "72454          Bdellovibrio bacteriovorus         Bdellovibrio bacteriovorus   \n",
       "72458          Bdellovibrio bacteriovorus         Bdellovibrio bacteriovorus   \n",
       "72459               Bacteriovorax stolpii              Bacteriovorax stolpii   \n",
       "...                                   ...                                ...   \n",
       "647687          Gemmatimonas groenlandica          Gemmatimonas groenlandica   \n",
       "647896        Chthonomonadaceae bacterium        Chthonomonadaceae bacterium   \n",
       "648315              Bdellovibrio sp. KM01              Bdellovibrio sp. KM01   \n",
       "648637         Chloroflexi bacterium G233         Chloroflexi bacterium G233   \n",
       "650081  Candidatus Atribacteria bacterium  Candidatus Atribacteria bacterium   \n",
       "\n",
       "          domain              phylum             class               order  \\\n",
       "72261   Bacteria  Desulfobacterota_I  Desulfovibrionia  Desulfovibrionales   \n",
       "72354   Bacteria  Desulfobacterota_I  Desulfovibrionia  Desulfovibrionales   \n",
       "72454   Bacteria    Bdellovibrionota   Bdellovibrionia   Bdellovibrionales   \n",
       "72458   Bacteria    Bdellovibrionota   Bdellovibrionia   Bdellovibrionales   \n",
       "72459   Bacteria    Bdellovibrionota   Bacteriovoracia   Bacteriovoracales   \n",
       "...          ...                 ...               ...                 ...   \n",
       "647687  Bacteria     Gemmatimonadota  Gemmatimonadetes    Gemmatimonadales   \n",
       "647896  Bacteria      Armatimonadota    Fimbriimonadia    Fimbriimonadales   \n",
       "648315  Bacteria    Bdellovibrionota   Bdellovibrionia   Bdellovibrionales   \n",
       "648637  Bacteria       Chloroflexota   Dehalococcoidia      Tepidiformales   \n",
       "650081  Bacteria       Atribacterota      Atribacteria      Atribacterales   \n",
       "\n",
       "                     family          genus                        species  \n",
       "72261   Desulfovibrionaceae  Desulfovibrio  Desulfovibrio desulfuricans_C  \n",
       "72354   Desulfovibrionaceae  Desulfovibrio          Desulfovibrio piger_A  \n",
       "72454    Bdellovibrionaceae   Bdellovibrio     Bdellovibrio bacteriovorus  \n",
       "72458    Bdellovibrionaceae   Bdellovibrio   Bdellovibrio bacteriovorus_C  \n",
       "72459    Bacteriovoracaceae  Bacteriovorax          Bacteriovorax stolpii  \n",
       "...                     ...            ...                            ...  \n",
       "647687    Gemmatimonadaceae   Gemmatimonas       Gemmatimonas sp013004105  \n",
       "647896    Fimbriimonadaceae        H1-ARM1            H1-ARM1 sp001567425  \n",
       "648315   Bdellovibrionaceae   Bdellovibrio       Bdellovibrio sp013752535  \n",
       "648637      Tepidiformaceae    Tepidiforma        Tepidiforma sp002563855  \n",
       "650081      Atribacteraceae     Atribacter           Atribacter laminatus  \n",
       "\n",
       "[121 rows x 41 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "0abcd288",
   "metadata": {},
   "outputs": [],
   "source": [
    "# list(mergedf[mergedf['assembly_accession'].isin(genomesofinterestList)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "252f775a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>phispy contig</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>length</th>\n",
       "      <th>ngenes</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_004801255.1</td>\n",
       "      <td>CP036295.1</td>\n",
       "      <td>2153716</td>\n",
       "      <td>2198232</td>\n",
       "      <td>44517</td>\n",
       "      <td>38</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_002872415.1</td>\n",
       "      <td>CP025704.1</td>\n",
       "      <td>1158874</td>\n",
       "      <td>1234669</td>\n",
       "      <td>75796</td>\n",
       "      <td>96</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_002872415.1</td>\n",
       "      <td>CP025704.1</td>\n",
       "      <td>3657103</td>\n",
       "      <td>3711357</td>\n",
       "      <td>54255</td>\n",
       "      <td>39</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_001273775.1</td>\n",
       "      <td>CP011801.1</td>\n",
       "      <td>1130180</td>\n",
       "      <td>1184283</td>\n",
       "      <td>54104</td>\n",
       "      <td>57</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000196815.1</td>\n",
       "      <td>FP929003.1</td>\n",
       "      <td>3819445</td>\n",
       "      <td>3875538</td>\n",
       "      <td>56094</td>\n",
       "      <td>66</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCA_002951815.1</td>\n",
       "      <td>CP019454.1</td>\n",
       "      <td>56295</td>\n",
       "      <td>110436</td>\n",
       "      <td>54142</td>\n",
       "      <td>62</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCA_000020485.1</td>\n",
       "      <td>CP001098.1</td>\n",
       "      <td>1774084</td>\n",
       "      <td>1832770</td>\n",
       "      <td>58687</td>\n",
       "      <td>36</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GCA_000017805.1</td>\n",
       "      <td>CP000804.1</td>\n",
       "      <td>5171089</td>\n",
       "      <td>5214761</td>\n",
       "      <td>43673</td>\n",
       "      <td>32</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCA_000017805.1</td>\n",
       "      <td>CP000804.1</td>\n",
       "      <td>1775718</td>\n",
       "      <td>1816184</td>\n",
       "      <td>40467</td>\n",
       "      <td>31</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCA_000023225.1</td>\n",
       "      <td>CP001629.1</td>\n",
       "      <td>2849308</td>\n",
       "      <td>2934306</td>\n",
       "      <td>84999</td>\n",
       "      <td>80</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GCA_000144695.1</td>\n",
       "      <td>CP002105.1</td>\n",
       "      <td>1831095</td>\n",
       "      <td>1881221</td>\n",
       "      <td>50127</td>\n",
       "      <td>59</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCA_000724625.1</td>\n",
       "      <td>CP007139.1</td>\n",
       "      <td>619098</td>\n",
       "      <td>676928</td>\n",
       "      <td>57831</td>\n",
       "      <td>51</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GCA_000724625.1</td>\n",
       "      <td>CP007139.1</td>\n",
       "      <td>1303179</td>\n",
       "      <td>1344216</td>\n",
       "      <td>41038</td>\n",
       "      <td>46</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GCA_000091165.1</td>\n",
       "      <td>FP565575.1</td>\n",
       "      <td>1890297</td>\n",
       "      <td>1954371</td>\n",
       "      <td>64075</td>\n",
       "      <td>84</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GCA_000091165.1</td>\n",
       "      <td>FP565575.1</td>\n",
       "      <td>444534</td>\n",
       "      <td>490528</td>\n",
       "      <td>45995</td>\n",
       "      <td>53</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GCA_000241815.1</td>\n",
       "      <td>CM001402.1</td>\n",
       "      <td>3394564</td>\n",
       "      <td>3448247</td>\n",
       "      <td>53684</td>\n",
       "      <td>34</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GCA_001886815.1</td>\n",
       "      <td>CP018099.1</td>\n",
       "      <td>1726762</td>\n",
       "      <td>1773528</td>\n",
       "      <td>46767</td>\n",
       "      <td>39</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GCA_000266925.1</td>\n",
       "      <td>CP003198.1</td>\n",
       "      <td>583181</td>\n",
       "      <td>633724</td>\n",
       "      <td>50544</td>\n",
       "      <td>49</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GCA_000299235.1</td>\n",
       "      <td>CP002919.1</td>\n",
       "      <td>1637283</td>\n",
       "      <td>1690298</td>\n",
       "      <td>53016</td>\n",
       "      <td>66</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GCA_007827295.1</td>\n",
       "      <td>VISF01000001.1</td>\n",
       "      <td>2193169</td>\n",
       "      <td>2248405</td>\n",
       "      <td>55237</td>\n",
       "      <td>61</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GCA_000284315.1</td>\n",
       "      <td>AP012342.1</td>\n",
       "      <td>842663</td>\n",
       "      <td>885919</td>\n",
       "      <td>43257</td>\n",
       "      <td>20</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCA_001007995.1</td>\n",
       "      <td>CP011389.1</td>\n",
       "      <td>1697187</td>\n",
       "      <td>1751500</td>\n",
       "      <td>54314</td>\n",
       "      <td>49</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GCA_003991135.1</td>\n",
       "      <td>CP016379.1</td>\n",
       "      <td>1060522</td>\n",
       "      <td>1114596</td>\n",
       "      <td>54075</td>\n",
       "      <td>52</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GCA_003991135.1</td>\n",
       "      <td>CP016379.1</td>\n",
       "      <td>2409109</td>\n",
       "      <td>2463683</td>\n",
       "      <td>54575</td>\n",
       "      <td>43</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GCA_900169565.1</td>\n",
       "      <td>LT828648.1</td>\n",
       "      <td>2191819</td>\n",
       "      <td>2241586</td>\n",
       "      <td>49768</td>\n",
       "      <td>49</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GCA_001544015.1</td>\n",
       "      <td>AP014924.1</td>\n",
       "      <td>1721603</td>\n",
       "      <td>1787775</td>\n",
       "      <td>66173</td>\n",
       "      <td>50</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GCA_001458695.1</td>\n",
       "      <td>LN885086.1</td>\n",
       "      <td>1658845</td>\n",
       "      <td>1726293</td>\n",
       "      <td>67449</td>\n",
       "      <td>69</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GCA_001458695.1</td>\n",
       "      <td>LN885086.1</td>\n",
       "      <td>2218292</td>\n",
       "      <td>2264009</td>\n",
       "      <td>45718</td>\n",
       "      <td>38</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GCA_011405515.1</td>\n",
       "      <td>AP022671.1</td>\n",
       "      <td>817807</td>\n",
       "      <td>859584</td>\n",
       "      <td>41778</td>\n",
       "      <td>41</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GCA_002162095.1</td>\n",
       "      <td>CP021424.1</td>\n",
       "      <td>2584357</td>\n",
       "      <td>2633422</td>\n",
       "      <td>49066</td>\n",
       "      <td>68</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GCA_013693795.1</td>\n",
       "      <td>CP059258.1</td>\n",
       "      <td>540747</td>\n",
       "      <td>605320</td>\n",
       "      <td>64574</td>\n",
       "      <td>81</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GCA_013693795.1</td>\n",
       "      <td>CP059258.1</td>\n",
       "      <td>4503354</td>\n",
       "      <td>4545701</td>\n",
       "      <td>42348</td>\n",
       "      <td>53</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GCA_013693795.1</td>\n",
       "      <td>CP059258.1</td>\n",
       "      <td>2298870</td>\n",
       "      <td>2338997</td>\n",
       "      <td>40128</td>\n",
       "      <td>51</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GCA_016699655.1</td>\n",
       "      <td>CP064986.1</td>\n",
       "      <td>4684786</td>\n",
       "      <td>4735542</td>\n",
       "      <td>50757</td>\n",
       "      <td>41</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>GCA_900498245.1</td>\n",
       "      <td>LS998015.1</td>\n",
       "      <td>2148368</td>\n",
       "      <td>2202840</td>\n",
       "      <td>54473</td>\n",
       "      <td>51</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>GCA_900498245.1</td>\n",
       "      <td>LS998015.1</td>\n",
       "      <td>2031498</td>\n",
       "      <td>2097300</td>\n",
       "      <td>65803</td>\n",
       "      <td>46</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>GCA_900498245.1</td>\n",
       "      <td>LS998015.1</td>\n",
       "      <td>2803169</td>\n",
       "      <td>2847929</td>\n",
       "      <td>44761</td>\n",
       "      <td>39</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GCA_014701195.1</td>\n",
       "      <td>AP018712.1</td>\n",
       "      <td>2015190</td>\n",
       "      <td>2056770</td>\n",
       "      <td>41581</td>\n",
       "      <td>37</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>GCA_003290465.1</td>\n",
       "      <td>CP030759.1</td>\n",
       "      <td>395323</td>\n",
       "      <td>454790</td>\n",
       "      <td>59468</td>\n",
       "      <td>40</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>GCA_014058405.1</td>\n",
       "      <td>CP047423.1</td>\n",
       "      <td>124920</td>\n",
       "      <td>214252</td>\n",
       "      <td>89333</td>\n",
       "      <td>108</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>GCA_014058405.1</td>\n",
       "      <td>CP047423.1</td>\n",
       "      <td>2600355</td>\n",
       "      <td>2652764</td>\n",
       "      <td>52410</td>\n",
       "      <td>59</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GCA_014058405.1</td>\n",
       "      <td>CP047423.1</td>\n",
       "      <td>824088</td>\n",
       "      <td>873050</td>\n",
       "      <td>48963</td>\n",
       "      <td>46</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>GCA_902702915.1</td>\n",
       "      <td>LR738849.1</td>\n",
       "      <td>2609577</td>\n",
       "      <td>2659276</td>\n",
       "      <td>49700</td>\n",
       "      <td>48</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>GCA_902809825.1</td>\n",
       "      <td>LR778114.1</td>\n",
       "      <td>1796952</td>\n",
       "      <td>1852104</td>\n",
       "      <td>55153</td>\n",
       "      <td>74</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>GCA_013004105.1</td>\n",
       "      <td>CP053085.1</td>\n",
       "      <td>3188692</td>\n",
       "      <td>3230773</td>\n",
       "      <td>42082</td>\n",
       "      <td>27</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>GCA_015775515.1</td>\n",
       "      <td>CP065383.1</td>\n",
       "      <td>501880</td>\n",
       "      <td>555083</td>\n",
       "      <td>53204</td>\n",
       "      <td>54</td>\n",
       "      <td>Dropped. No genes were identified as phage genes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   assembly_accession   phispy contig    start     stop  length  ngenes  \\\n",
       "0     GCA_004801255.1      CP036295.1  2153716  2198232   44517      38   \n",
       "1     GCA_002872415.1      CP025704.1  1158874  1234669   75796      96   \n",
       "2     GCA_002872415.1      CP025704.1  3657103  3711357   54255      39   \n",
       "3     GCA_001273775.1      CP011801.1  1130180  1184283   54104      57   \n",
       "4     GCA_000196815.1      FP929003.1  3819445  3875538   56094      66   \n",
       "5     GCA_002951815.1      CP019454.1    56295   110436   54142      62   \n",
       "6     GCA_000020485.1      CP001098.1  1774084  1832770   58687      36   \n",
       "7     GCA_000017805.1      CP000804.1  5171089  5214761   43673      32   \n",
       "8     GCA_000017805.1      CP000804.1  1775718  1816184   40467      31   \n",
       "9     GCA_000023225.1      CP001629.1  2849308  2934306   84999      80   \n",
       "10    GCA_000144695.1      CP002105.1  1831095  1881221   50127      59   \n",
       "11    GCA_000724625.1      CP007139.1   619098   676928   57831      51   \n",
       "12    GCA_000724625.1      CP007139.1  1303179  1344216   41038      46   \n",
       "13    GCA_000091165.1      FP565575.1  1890297  1954371   64075      84   \n",
       "14    GCA_000091165.1      FP565575.1   444534   490528   45995      53   \n",
       "15    GCA_000241815.1      CM001402.1  3394564  3448247   53684      34   \n",
       "16    GCA_001886815.1      CP018099.1  1726762  1773528   46767      39   \n",
       "17    GCA_000266925.1      CP003198.1   583181   633724   50544      49   \n",
       "18    GCA_000299235.1      CP002919.1  1637283  1690298   53016      66   \n",
       "19    GCA_007827295.1  VISF01000001.1  2193169  2248405   55237      61   \n",
       "20    GCA_000284315.1      AP012342.1   842663   885919   43257      20   \n",
       "21    GCA_001007995.1      CP011389.1  1697187  1751500   54314      49   \n",
       "22    GCA_003991135.1      CP016379.1  1060522  1114596   54075      52   \n",
       "23    GCA_003991135.1      CP016379.1  2409109  2463683   54575      43   \n",
       "24    GCA_900169565.1      LT828648.1  2191819  2241586   49768      49   \n",
       "25    GCA_001544015.1      AP014924.1  1721603  1787775   66173      50   \n",
       "26    GCA_001458695.1      LN885086.1  1658845  1726293   67449      69   \n",
       "27    GCA_001458695.1      LN885086.1  2218292  2264009   45718      38   \n",
       "28    GCA_011405515.1      AP022671.1   817807   859584   41778      41   \n",
       "29    GCA_002162095.1      CP021424.1  2584357  2633422   49066      68   \n",
       "30    GCA_013693795.1      CP059258.1   540747   605320   64574      81   \n",
       "31    GCA_013693795.1      CP059258.1  4503354  4545701   42348      53   \n",
       "32    GCA_013693795.1      CP059258.1  2298870  2338997   40128      51   \n",
       "33    GCA_016699655.1      CP064986.1  4684786  4735542   50757      41   \n",
       "34    GCA_900498245.1      LS998015.1  2148368  2202840   54473      51   \n",
       "35    GCA_900498245.1      LS998015.1  2031498  2097300   65803      46   \n",
       "36    GCA_900498245.1      LS998015.1  2803169  2847929   44761      39   \n",
       "37    GCA_014701195.1      AP018712.1  2015190  2056770   41581      37   \n",
       "38    GCA_003290465.1      CP030759.1   395323   454790   59468      40   \n",
       "39    GCA_014058405.1      CP047423.1   124920   214252   89333     108   \n",
       "40    GCA_014058405.1      CP047423.1  2600355  2652764   52410      59   \n",
       "41    GCA_014058405.1      CP047423.1   824088   873050   48963      46   \n",
       "42    GCA_902702915.1      LR738849.1  2609577  2659276   49700      48   \n",
       "43    GCA_902809825.1      LR778114.1  1796952  1852104   55153      74   \n",
       "44    GCA_013004105.1      CP053085.1  3188692  3230773   42082      27   \n",
       "45    GCA_015775515.1      CP065383.1   501880   555083   53204      54   \n",
       "\n",
       "                                               notes  \n",
       "0   Dropped. No genes were identified as phage genes  \n",
       "1   Dropped. No genes were identified as phage genes  \n",
       "2   Dropped. No genes were identified as phage genes  \n",
       "3   Dropped. No genes were identified as phage genes  \n",
       "4   Dropped. No genes were identified as phage genes  \n",
       "5   Dropped. No genes were identified as phage genes  \n",
       "6   Dropped. No genes were identified as phage genes  \n",
       "7   Dropped. No genes were identified as phage genes  \n",
       "8   Dropped. No genes were identified as phage genes  \n",
       "9   Dropped. No genes were identified as phage genes  \n",
       "10  Dropped. No genes were identified as phage genes  \n",
       "11  Dropped. No genes were identified as phage genes  \n",
       "12  Dropped. No genes were identified as phage genes  \n",
       "13  Dropped. No genes were identified as phage genes  \n",
       "14  Dropped. No genes were identified as phage genes  \n",
       "15  Dropped. No genes were identified as phage genes  \n",
       "16  Dropped. No genes were identified as phage genes  \n",
       "17  Dropped. No genes were identified as phage genes  \n",
       "18  Dropped. No genes were identified as phage genes  \n",
       "19  Dropped. No genes were identified as phage genes  \n",
       "20  Dropped. No genes were identified as phage genes  \n",
       "21  Dropped. No genes were identified as phage genes  \n",
       "22  Dropped. No genes were identified as phage genes  \n",
       "23  Dropped. No genes were identified as phage genes  \n",
       "24  Dropped. No genes were identified as phage genes  \n",
       "25  Dropped. No genes were identified as phage genes  \n",
       "26  Dropped. No genes were identified as phage genes  \n",
       "27  Dropped. No genes were identified as phage genes  \n",
       "28  Dropped. No genes were identified as phage genes  \n",
       "29  Dropped. No genes were identified as phage genes  \n",
       "30  Dropped. No genes were identified as phage genes  \n",
       "31  Dropped. No genes were identified as phage genes  \n",
       "32  Dropped. No genes were identified as phage genes  \n",
       "33  Dropped. No genes were identified as phage genes  \n",
       "34  Dropped. No genes were identified as phage genes  \n",
       "35  Dropped. No genes were identified as phage genes  \n",
       "36  Dropped. No genes were identified as phage genes  \n",
       "37  Dropped. No genes were identified as phage genes  \n",
       "38  Dropped. No genes were identified as phage genes  \n",
       "39  Dropped. No genes were identified as phage genes  \n",
       "40  Dropped. No genes were identified as phage genes  \n",
       "41  Dropped. No genes were identified as phage genes  \n",
       "42  Dropped. No genes were identified as phage genes  \n",
       "43  Dropped. No genes were identified as phage genes  \n",
       "44  Dropped. No genes were identified as phage genes  \n",
       "45  Dropped. No genes were identified as phage genes  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "interrogateSubsetDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "35c6f4a0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:prophage-distributions] *",
   "language": "python",
   "name": "conda-env-prophage-distributions-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}